packages/rich-text/src/plugins/List/index.tsx

```diff
@@ -149,6 +152,60 @@ export const withListOptions: CustomSlatePluginOptions = {
   },
 };
 
+const emptyNodeOfType = (type) => ({ type, children: [], data: {} });
+
+const isList = (node: CustomElement) =>
+  [BLOCKS.OL_LIST, BLOCKS.UL_LIST].includes(node.type as BLOCKS);
+
+const isListItem = (node: CustomElement) => node.type === BLOCKS.LIST_ITEM;
+
+const hasListAsDirectParent = (editor: Editor, path: Path) => {
+  const [parentNode] = (getParent(editor, path) || []) as NodeEntry;
+  return isList(parentNode as CustomElement);
+};
+
+const isValidInsideList = (node: CustomElement) =>
+  (LIST_ITEM_BLOCKS as Array<TopLevelBlockEnum | INLINES>)
+    .concat(Object.values(INLINES))
+    .includes(node.type as TopLevelBlockEnum);
+
+const unwrapInvalidGrandchildren = (editor: Editor, path: Path) => {
+  for (const [grandchild, grandchildPath] of Node.children(editor, path)) {
+    if (!isValidInsideList(grandchild as CustomElement)) {
+      Transforms.unwrapNodes(editor, { at: grandchildPath });
```

Comment:
I don't understand the expected behavior in this case. I thought we are supposed to convert the invalid nodes to text?

Or are you waiting for the helper to be available?

Reply:
Yep, ideally we'd be calling the to-text helper here. Should have thrown a TODO in there